File: viikko2/c++/helloworld/Harjoittelija.cpp
#include "Harjoittelija.h"

Harjoittelija::Harjoittelija() : Opiskelija(), mTyopaikka("tuntematon"), mPalkka(0)
{
    cout << "Harjoittelijan oletusrakentaja" << endl;
}

Harjoittelija::Harjoittelija(const string& aNimi, const string& aOpiskelijanumero, const string& aTyopaikka, float aPalkka)
    : Opiskelija(aNimi, aOpiskelijanumero), mTyopaikka(aTyopaikka), mPalkka(aPalkka)
{
    cout << "Harjoittelijan 4 parametrinen rakentaja " << endl;
}

Harjoittelija::~Harjoittelija()
{
    cout <<"Opiskelija " << getNimi() << " vapautettu muistista" << endl;
}


//ToDo getterit ja setterit

void Harjoittelija::setTyopaikka( const string& aTyopaikka)
{
    mTyopaikka = aTyopaikka;
}
void Harjoittelija::setPalkka(float aPalkka)
{
    mPalkka = aPalkka;
}

void Harjoittelija::tulostaTiedot() const
{
    // voidaan tylostaa myös ensin kantaluokan tulostus, esim. Opikselija::tulostaTiedot();
    cout << "Nimi: " << getNimi() << endl;
    cout <<"Opintopisteet: " << getOpintopisteet() <<endl;
    cout << "Harjoittelupaikka: " << mTyopaikka <<endl;
    cout << "Palkka: " << mPalkka << endl;
}

File: viikko4/java/old/Harjoittelija.java
public class Harjoittelija extends Opiskelija {
    private String mTyopaikka;
    private float mPalkka;

    public Harjoittelija(){
        mTyopaikka = "Tuntematon";
        mPalkka = 0;
    }
    public Harjoittelija(String aNimi, String aOpiskelijanumero, String aTyopaikka, float aPalkka){
        super(aNimi, aOpiskelijanumero);
        mTyopaikka = aTyopaikka;
        mPalkka = aPalkka;
        System.out.println("Harjoittelijan 4 parametrinen rakentaja");


    }
    public void setTyopaikka(String aTyopaikka){
        mTyopaikka = aTyopaikka;
    }

    public String getTyopaikka(){
        return mTyopaikka;
    }
    public void setPalkka(float aPalkka){
        mPalkka = aPalkka;
    }

    public float getPalkka(){
        return mPalkka;
    }



    public void tulostaTiedot() {
        System.out.println("Nimi: " + getNimi());
        System.out.println("Opintopisteet: " + getOpintopisteet());
        System.out.println("Palkka: " + mPalkka); //tässä olisi myös hyvä käyttää getteriä!

    }
}
